dynamiq/storages/vector/elasticsearch/elasticsearch.py

```diff
@@ -0,0 +1,377 @@
+from enum import Enum
+from typing import Any
+
+import numpy as np
+from elasticsearch import NotFoundError
+
+from dynamiq.connections import Elasticsearch
+from dynamiq.storages.vector.base import BaseVectorStoreParams, BaseWriterVectorStoreParams
+from dynamiq.storages.vector.exceptions import VectorStoreException
+from dynamiq.storages.vector.policies import DuplicatePolicy
+from dynamiq.types import Document
+from dynamiq.utils.logger import logger
+
+
+class ElasticsearchSimilarityMetric(str, Enum):
+    """Supported similarity metrics for Elasticsearch."""
+
+    COSINE = "cosine"
+    DOT_PRODUCT = "dot_product"
+    L2 = "l2"
+
+
+class ElasticsearchVectorStoreParams(BaseVectorStoreParams):
+    """Parameters for Elasticsearch vector store.
+
+    Attributes:
+        index_name (str): Name of the index. Defaults to "default".
+        dimension (int): Dimension of the vectors. Defaults to 1536.
+        similarity (str): Similarity metric to use. Defaults to "cosine".
+        content_key (str): Key for content field. Defaults to "content".
+        embedding_key (str): Key for embedding field. Defaults to "embedding".
+    """
+
+    dimension: int = 1536
+    similarity: ElasticsearchSimilarityMetric = ElasticsearchSimilarityMetric.COSINE
+    embedding_key: str = "embedding"
+    write_batch_size: int = 100
+    scroll_size: int = 1000
+
+
+class ElasticsearchVectorStoreWriterParams(ElasticsearchVectorStoreParams, BaseWriterVectorStoreParams):
+    """Parameters for Elasticsearch vector store writer."""
+
+    pass
+
+
+class ElasticsearchVectorStore:
+    """Vector store using Elasticsearch for dense vector search."""
+
+    def __init__(
+        self,
+        connection: Elasticsearch | None = None,
+        client: Elasticsearch | None = None,
```

Comment:
```suggestion
        connection: ElasticsearchConnection | None = None,
        client: Elasticsearch | None = None,
```
Is the Elasticsearch client a Connection here? If not, let's separate them and ensure proper typing.